pkg/kubelet/kubelet_pods.go

```diff
@@ -687,11 +687,7 @@ func (kl *Kubelet) podIsTerminated(pod *v1.Pod) bool {
 		// restarted.
 		status = pod.Status
 	}
-	if status.Phase == v1.PodFailed || status.Phase == v1.PodSucceeded {
-		return true
-	}
-
-	return false
+	return status.Phase == v1.PodFailed || status.Phase == v1.PodSucceeded || (pod.DeletionTimestamp != nil && notRunning(status.ContainerStatuses))
```

Comment:
@derekwaynecarr your change to `podIsTerminated` may cause other race conditions (e.g., pod worker still trying to start new containers). Could you remove this? 

/cc @Random-Liu @dchen1107